modules/alookup/alookup.go

```diff
@@ -68,11 +81,16 @@ func (s *Lookup) doLookupProtocol(name, nameServer string, dnsType uint16, candi
 			// filter only valid answers of requested type or CNAME (#163)
 			if ans, ok := a.(miekg.Answer); ok {
 				lowerCaseName := strings.ToLower(ans.Name)
-				ansType := dns.StringToType[ans.Type]
-				if dnsType == ansType {
-					candidateSet[lowerCaseName] = append(candidateSet[lowerCaseName], ans)
-				} else if ok && dns.TypeCNAME == ansType {
-					cnameSet[lowerCaseName] = append(cnameSet[lowerCaseName], ans)
+				// Verify that the answer type matches requested type
```

Comment:
Only thing I might bring up is that we've got some duplication going on here with what happens in each loop, it may be cleaner to pull that logic out. Thoughts?

Reply:
Is the answer always an A/AAAA/CNAME record? What if you query for a NS record? Does that go through a different logic?